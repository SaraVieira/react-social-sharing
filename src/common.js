import styled, { css } from 'styled-components'
import is, { isOr } from 'styled-is'

export const SharingButton = styled.div`
  border-radius: 5px;
  transition: 25ms ease-out;
  padding: 0.5em 0.75em;
  font-family: Helvetica Neue, Helvetica, Arial, sans-serif;
`

export const Link = styled.a`
  text-decoration: none;
  color: #fff;
  margin: 0.5em;
  display: inline-block;

  svg {
    width: 1em;
    height: 1em;
    vertical-align: top;

    &:not(:only-child) {
      margin-right: 0.4em;
    }
  }

  ${is('small')`
    svg {
      vertical-align: middle;
    }
  `};

  ${is('simple')`
    svg {
      stroke: #232323;
      fill: none;
    }
  `};

  ${is('simpleReverse')`
    svg {
      stroke: #fff;
      fill: none;
    }
  `};
`

export const Icon = styled.div`
  display: flex;
  align-items: center;
  fill: #fff;
  stroke: none;

  ${isOr('solid', 'solidcircle')`
    fill: #fff;
    stroke: none
  `};
`
