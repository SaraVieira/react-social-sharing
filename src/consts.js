export default {
  twitter: (link = '', message = '') =>
    `https://twitter.com/intent/tweet/?text=${encodeURIComponent(
      message
    )}&url=${encodeURIComponent(link)}`,
  facebook: (link: '') =>
    `https://facebook.com/sharer/sharer.php?u=${encodeURIComponent(link)}`,
  google: (link: '') =>
    `https://plus.google.com/share?url=${encodeURIComponent(link)}`,
  tumblr: (link: '') =>
    `http://tumblr.com/widgets/share/tool?canonicalUrl=${encodeURIComponent(
      link
    )}`,
  reddit: (link: '') =>
    `https://reddit.com/submit/?url=${encodeURIComponent(link)}`,
  whatsapp: (link = '', message = '') =>
    `https://api.whatsapp.com/send?text=${encodeURIComponent(message)}%20${encodeURIComponent(
      link
    )}`,
  telegram: (link = '', message = '') =>
    `https://telegram.me/share/url?text=${encodeURIComponent(
      message
    )}&url=${encodeURIComponent(link)}`,
  vk: (link = '', message = '') =>
    `http://vk.com/share.php?title=${encodeURIComponent(
      message
    )}&url=${encodeURIComponent(link)}`,
  hacker: (link = '', message = '') =>
    `https://news.ycombinator.com/submitlink?u=${encodeURIComponent(
      link
    )}&t=${encodeURIComponent(message)}`,
  xing: (link = '', message = '') =>
    `https://www.xing.com/app/user?op=share;url=${encodeURIComponent(
      link
    )};title=${encodeURIComponent(message)}.`,
  mail: (link = '', subject, body) =>
    `mailto:?subject=${encodeURIComponent(
      subject || ''
    )}&body=${encodeURIComponent((body && `${body}\n\n${link}`) || link)}`,
  pinterest: (link = '', message = '') =>
    `https://pinterest.com/pin/create/button/?url=${encodeURIComponent(
      link
    )}&media=${encodeURIComponent(link)}&description=${encodeURIComponent(
      message
    )}`,
  linkedin: (link = '', message = '') =>
    `https://www.linkedin.com/shareArticle?mini=true&url=${encodeURIComponent(
      link
    )}&title=${encodeURIComponent(message)}&summary=${encodeURIComponent(
      message
    )}&source=${encodeURIComponent(link)}`
}
