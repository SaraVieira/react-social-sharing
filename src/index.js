// this should be the entry point to your library
module.exports = {
  Twitter: require('./buttons/twitter').default,
  Facebook: require('./buttons/facebook').default,
  Google: require('./buttons/google').default,
  Tumblr: require('./buttons/tumblr').default,
  Mail: require('./buttons/mail').default,
  Pinterest: require('./buttons/pinterest').default,
  Linkedin: require('./buttons/linkedin').default,
  Reddit: require('./buttons/reddit').default,
  Xing: require('./buttons/xing').default,
  Whatsapp: require('./buttons/whatsapp').default,
  HackerNews: require('./buttons/hacker').default,
  VK: require('./buttons/vk').default,
  Telegram: require('./buttons/telegram').default,
  X: require('./buttons/x').default
}
