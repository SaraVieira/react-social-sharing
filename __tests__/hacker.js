import React from 'react'
import { HackerNews } from '../src/index'
import 'jest-styled-components'

describe('HackerNews', function() {
  test('1', () => {
    const wrapper = mount(
      <HackerNews
        solidcircle
        big
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('2', () => {
    const wrapper = mount(
      <HackerNews
        solidcircle
        medium
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('3', () => {
    const wrapper = mount(
      <HackerNews
        solidcircle
        small
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('4', () => {
    const wrapper = mount(
      <HackerNews
        circle
        big
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('5', () => {
    const wrapper = mount(
      <HackerNews
        circle
        medium
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('6', () => {
    const wrapper = mount(
      <HackerNews
        circle
        small
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('7', () => {
    const wrapper = mount(
      <HackerNews
        solid
        big
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('8', () => {
    const wrapper = mount(
      <HackerNews
        solid
        medium
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('9', () => {
    const wrapper = mount(
      <HackerNews
        solid
        small
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('11', () => {
    const wrapper = mount(<HackerNews circle />)
    expect(wrapper).toMatchSnapshot()
  })
  test('11', () => {
    const wrapper = mount(<HackerNews solidcircle />)
    expect(wrapper).toMatchSnapshot()
  })
  test('12', () => {
    const wrapper = mount(<HackerNews />)
    expect(wrapper).toMatchSnapshot()
  })
  test('13', () => {
    const wrapper = mount(<HackerNews simple />)
    expect(wrapper).toMatchSnapshot()
  })
  test('14', () => {
    const wrapper = mount(<HackerNews simpleReverse />)
    expect(wrapper).toMatchSnapshot()
  })
})
