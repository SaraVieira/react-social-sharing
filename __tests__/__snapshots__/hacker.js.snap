// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HackerNews 1 1`] = `
<Button
  big={true}
  link="http://sharingbuttons.io"
  message="I am so cool"
  solidcircle={true}
>
  <styled.a
    aria-label="Share on HackerNews"
    href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on HackerNews"
      className="sc-bwzfXH ewcqji"
      href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-VigVT bgFikP"
        >
          <styled.div
            aria-hidden="true"
            solidcircle={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 256 256"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 256c70.692 0 128-57.308 128-128C256 57.308 198.692 0 128 0 57.308 0 0 57.308 0 128c0 70.692 57.308 128 128 128zm-9.06-113.686L75 60h20.08l25.85 52.093c.397.927.86 1.888 1.39 2.883.53.994.995 2.02 1.393 3.08.265.4.463.764.596 1.095.13.334.262.63.395.898.662 1.325 1.26 2.618 1.79 3.877.53 1.26.993 2.42 1.39 3.48 1.06-2.254 2.22-4.673 3.48-7.258 1.26-2.585 2.552-5.27 3.877-8.052L161.49 60h18.69l-44.34 83.308v53.087h-16.9v-54.08z"
                  fillRule="evenodd"
                />
              </svg>
               
              <span>
                share on HackerNews
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`HackerNews 2 1`] = `
<Button
  link="http://sharingbuttons.io"
  medium={true}
  message="I am so cool"
  solidcircle={true}
>
  <styled.a
    aria-label="Share on HackerNews"
    href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on HackerNews"
      className="sc-bwzfXH ewcqji"
      href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-VigVT bgFikP"
        >
          <styled.div
            aria-hidden="true"
            solidcircle={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 256 256"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 256c70.692 0 128-57.308 128-128C256 57.308 198.692 0 128 0 57.308 0 0 57.308 0 128c0 70.692 57.308 128 128 128zm-9.06-113.686L75 60h20.08l25.85 52.093c.397.927.86 1.888 1.39 2.883.53.994.995 2.02 1.393 3.08.265.4.463.764.596 1.095.13.334.262.63.395.898.662 1.325 1.26 2.618 1.79 3.877.53 1.26.993 2.42 1.39 3.48 1.06-2.254 2.22-4.673 3.48-7.258 1.26-2.585 2.552-5.27 3.877-8.052L161.49 60h18.69l-44.34 83.308v53.087h-16.9v-54.08z"
                  fillRule="evenodd"
                />
              </svg>
               
              <span>
                HackerNews
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`HackerNews 3 1`] = `
<Button
  link="http://sharingbuttons.io"
  message="I am so cool"
  small={true}
  solidcircle={true}
>
  <styled.a
    aria-label="Share on HackerNews"
    href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on HackerNews"
      className="sc-bwzfXH ewcqji"
      href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div
        small={true}
      >
        <div
          className="sc-VigVT bgFikP"
        >
          <styled.div
            aria-hidden="true"
            solidcircle={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 256 256"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 256c70.692 0 128-57.308 128-128C256 57.308 198.692 0 128 0 57.308 0 0 57.308 0 128c0 70.692 57.308 128 128 128zm-9.06-113.686L75 60h20.08l25.85 52.093c.397.927.86 1.888 1.39 2.883.53.994.995 2.02 1.393 3.08.265.4.463.764.596 1.095.13.334.262.63.395.898.662 1.325 1.26 2.618 1.79 3.877.53 1.26.993 2.42 1.39 3.48 1.06-2.254 2.22-4.673 3.48-7.258 1.26-2.585 2.552-5.27 3.877-8.052L161.49 60h18.69l-44.34 83.308v53.087h-16.9v-54.08z"
                  fillRule="evenodd"
                />
              </svg>
               
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`HackerNews 4 1`] = `
<Button
  big={true}
  circle={true}
  link="http://sharingbuttons.io"
  message="I am so cool"
>
  <styled.a
    aria-label="Share on HackerNews"
    href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on HackerNews"
      className="sc-bwzfXH ewcqji"
      href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-VigVT bgFikP"
        >
          <styled.div
            aria-hidden="true"
          >
            <div
              aria-hidden="true"
              className="sc-htpNat dZQCuR"
            >
              <svg
                viewBox="0 0 256 256"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="128"
                  cy="128"
                  r="122.5"
                />
                <path
                  d="M128 256c70.692 0 128-57.308 128-128C256 57.308 198.692 0 128 0 57.308 0 0 57.308 0 128c0 70.692 57.308 128 128 128zm-9.06-113.686L75 60h20.08l25.85 52.093c.397.927.86 1.888 1.39 2.883.53.994.995 2.02 1.393 3.08.265.4.463.764.596 1.095.13.334.262.63.395.898.662 1.325 1.26 2.618 1.79 3.877.53 1.26.993 2.42 1.39 3.48 1.06-2.254 2.22-4.673 3.48-7.258 1.26-2.585 2.552-5.27 3.877-8.052L161.49 60h18.69l-44.34 83.308v53.087h-16.9v-54.08z"
                  fillRule="evenodd"
                  strokeWidth="10px"
                />
              </svg>
               
              <span>
                share on HackerNews
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`HackerNews 5 1`] = `
<Button
  circle={true}
  link="http://sharingbuttons.io"
  medium={true}
  message="I am so cool"
>
  <styled.a
    aria-label="Share on HackerNews"
    href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on HackerNews"
      className="sc-bwzfXH ewcqji"
      href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-VigVT bgFikP"
        >
          <styled.div
            aria-hidden="true"
          >
            <div
              aria-hidden="true"
              className="sc-htpNat dZQCuR"
            >
              <svg
                viewBox="0 0 256 256"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="128"
                  cy="128"
                  r="122.5"
                />
                <path
                  d="M128 256c70.692 0 128-57.308 128-128C256 57.308 198.692 0 128 0 57.308 0 0 57.308 0 128c0 70.692 57.308 128 128 128zm-9.06-113.686L75 60h20.08l25.85 52.093c.397.927.86 1.888 1.39 2.883.53.994.995 2.02 1.393 3.08.265.4.463.764.596 1.095.13.334.262.63.395.898.662 1.325 1.26 2.618 1.79 3.877.53 1.26.993 2.42 1.39 3.48 1.06-2.254 2.22-4.673 3.48-7.258 1.26-2.585 2.552-5.27 3.877-8.052L161.49 60h18.69l-44.34 83.308v53.087h-16.9v-54.08z"
                  fillRule="evenodd"
                  strokeWidth="10px"
                />
              </svg>
               
              <span>
                HackerNews
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`HackerNews 6 1`] = `
<Button
  circle={true}
  link="http://sharingbuttons.io"
  message="I am so cool"
  small={true}
>
  <styled.a
    aria-label="Share on HackerNews"
    href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on HackerNews"
      className="sc-bwzfXH ewcqji"
      href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div
        small={true}
      >
        <div
          className="sc-VigVT bgFikP"
        >
          <styled.div
            aria-hidden="true"
          >
            <div
              aria-hidden="true"
              className="sc-htpNat dZQCuR"
            >
              <svg
                viewBox="0 0 256 256"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="128"
                  cy="128"
                  r="122.5"
                />
                <path
                  d="M128 256c70.692 0 128-57.308 128-128C256 57.308 198.692 0 128 0 57.308 0 0 57.308 0 128c0 70.692 57.308 128 128 128zm-9.06-113.686L75 60h20.08l25.85 52.093c.397.927.86 1.888 1.39 2.883.53.994.995 2.02 1.393 3.08.265.4.463.764.596 1.095.13.334.262.63.395.898.662 1.325 1.26 2.618 1.79 3.877.53 1.26.993 2.42 1.39 3.48 1.06-2.254 2.22-4.673 3.48-7.258 1.26-2.585 2.552-5.27 3.877-8.052L161.49 60h18.69l-44.34 83.308v53.087h-16.9v-54.08z"
                  fillRule="evenodd"
                  strokeWidth="10px"
                />
              </svg>
               
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`HackerNews 7 1`] = `
<Button
  big={true}
  link="http://sharingbuttons.io"
  message="I am so cool"
  solid={true}
>
  <styled.a
    aria-label="Share on HackerNews"
    href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on HackerNews"
      className="sc-bwzfXH ewcqji"
      href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-VigVT bgFikP"
        >
          <styled.div
            aria-hidden="true"
            solid={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 140 140"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M60.94 82.314L17 0h20.08l25.85 52.093c.397.927.86 1.888 1.39 2.883.53.994.995 2.02 1.393 3.08.265.4.463.764.596 1.095.13.334.262.63.395.898.662 1.325 1.26 2.618 1.79 3.877.53 1.26.993 2.42 1.39 3.48 1.06-2.254 2.22-4.673 3.48-7.258 1.26-2.585 2.552-5.27 3.877-8.052L103.49 0h18.69L77.84 83.308v53.087h-16.9v-54.08z"
                  fillRule="evenodd"
                />
              </svg>
               
              <span>
                share on HackerNews
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`HackerNews 8 1`] = `
<Button
  link="http://sharingbuttons.io"
  medium={true}
  message="I am so cool"
  solid={true}
>
  <styled.a
    aria-label="Share on HackerNews"
    href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on HackerNews"
      className="sc-bwzfXH ewcqji"
      href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-VigVT bgFikP"
        >
          <styled.div
            aria-hidden="true"
            solid={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 140 140"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M60.94 82.314L17 0h20.08l25.85 52.093c.397.927.86 1.888 1.39 2.883.53.994.995 2.02 1.393 3.08.265.4.463.764.596 1.095.13.334.262.63.395.898.662 1.325 1.26 2.618 1.79 3.877.53 1.26.993 2.42 1.39 3.48 1.06-2.254 2.22-4.673 3.48-7.258 1.26-2.585 2.552-5.27 3.877-8.052L103.49 0h18.69L77.84 83.308v53.087h-16.9v-54.08z"
                  fillRule="evenodd"
                />
              </svg>
               
              <span>
                HackerNews
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`HackerNews 9 1`] = `
<Button
  link="http://sharingbuttons.io"
  message="I am so cool"
  small={true}
  solid={true}
>
  <styled.a
    aria-label="Share on HackerNews"
    href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on HackerNews"
      className="sc-bwzfXH ewcqji"
      href="https://news.ycombinator.com/submitlink?u=http%3A%2F%2Fsharingbuttons.io&amp;t=I%20am%20so%20cool"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div
        small={true}
      >
        <div
          className="sc-VigVT bgFikP"
        >
          <styled.div
            aria-hidden="true"
            solid={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 140 140"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M60.94 82.314L17 0h20.08l25.85 52.093c.397.927.86 1.888 1.39 2.883.53.994.995 2.02 1.393 3.08.265.4.463.764.596 1.095.13.334.262.63.395.898.662 1.325 1.26 2.618 1.79 3.877.53 1.26.993 2.42 1.39 3.48 1.06-2.254 2.22-4.673 3.48-7.258 1.26-2.585 2.552-5.27 3.877-8.052L103.49 0h18.69L77.84 83.308v53.087h-16.9v-54.08z"
                  fillRule="evenodd"
                />
              </svg>
               
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`HackerNews 11 1`] = `
<Button
  circle={true}
>
  <styled.a
    aria-label="Share on HackerNews"
    href="https://news.ycombinator.com/submitlink?u=&amp;t=undefined"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on HackerNews"
      className="sc-bwzfXH ewcqji"
      href="https://news.ycombinator.com/submitlink?u=&amp;t=undefined"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-VigVT bgFikP"
        >
          <styled.div
            aria-hidden="true"
          >
            <div
              aria-hidden="true"
              className="sc-htpNat dZQCuR"
            >
              <svg
                viewBox="0 0 256 256"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="128"
                  cy="128"
                  r="122.5"
                />
                <path
                  d="M128 256c70.692 0 128-57.308 128-128C256 57.308 198.692 0 128 0 57.308 0 0 57.308 0 128c0 70.692 57.308 128 128 128zm-9.06-113.686L75 60h20.08l25.85 52.093c.397.927.86 1.888 1.39 2.883.53.994.995 2.02 1.393 3.08.265.4.463.764.596 1.095.13.334.262.63.395.898.662 1.325 1.26 2.618 1.79 3.877.53 1.26.993 2.42 1.39 3.48 1.06-2.254 2.22-4.673 3.48-7.258 1.26-2.585 2.552-5.27 3.877-8.052L161.49 60h18.69l-44.34 83.308v53.087h-16.9v-54.08z"
                  fillRule="evenodd"
                  strokeWidth="10px"
                />
              </svg>
               
              <span>
                HackerNews
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`HackerNews 11 2`] = `
<Button
  solidcircle={true}
>
  <styled.a
    aria-label="Share on HackerNews"
    href="https://news.ycombinator.com/submitlink?u=&amp;t=undefined"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on HackerNews"
      className="sc-bwzfXH ewcqji"
      href="https://news.ycombinator.com/submitlink?u=&amp;t=undefined"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-VigVT bgFikP"
        >
          <styled.div
            aria-hidden="true"
            solidcircle={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 256 256"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M128 256c70.692 0 128-57.308 128-128C256 57.308 198.692 0 128 0 57.308 0 0 57.308 0 128c0 70.692 57.308 128 128 128zm-9.06-113.686L75 60h20.08l25.85 52.093c.397.927.86 1.888 1.39 2.883.53.994.995 2.02 1.393 3.08.265.4.463.764.596 1.095.13.334.262.63.395.898.662 1.325 1.26 2.618 1.79 3.877.53 1.26.993 2.42 1.39 3.48 1.06-2.254 2.22-4.673 3.48-7.258 1.26-2.585 2.552-5.27 3.877-8.052L161.49 60h18.69l-44.34 83.308v53.087h-16.9v-54.08z"
                  fillRule="evenodd"
                />
              </svg>
               
              <span>
                HackerNews
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`HackerNews 12 1`] = `
<Button>
  <styled.a
    aria-label="Share on HackerNews"
    href="https://news.ycombinator.com/submitlink?u=&amp;t=undefined"
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on HackerNews"
      className="sc-bwzfXH ewcqji"
      href="https://news.ycombinator.com/submitlink?u=&amp;t=undefined"
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-VigVT bgFikP"
        >
          <styled.div
            aria-hidden="true"
            solid={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 140 140"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M60.94 82.314L17 0h20.08l25.85 52.093c.397.927.86 1.888 1.39 2.883.53.994.995 2.02 1.393 3.08.265.4.463.764.596 1.095.13.334.262.63.395.898.662 1.325 1.26 2.618 1.79 3.877.53 1.26.993 2.42 1.39 3.48 1.06-2.254 2.22-4.673 3.48-7.258 1.26-2.585 2.552-5.27 3.877-8.052L103.49 0h18.69L77.84 83.308v53.087h-16.9v-54.08z"
                  fillRule="evenodd"
                />
              </svg>
               
              <span>
                HackerNews
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`HackerNews 13 1`] = `
<Button
  simple={true}
>
  <styled.a
    aria-label="Share on HackerNews"
    href="https://news.ycombinator.com/submitlink?u=&amp;t=undefined"
    rel="noreferrer noopener"
    simple={true}
    target="_blank"
  >
    <a
      aria-label="Share on HackerNews"
      className="sc-bwzfXH ljNxya"
      href="https://news.ycombinator.com/submitlink?u=&amp;t=undefined"
      rel="noreferrer noopener"
      target="_blank"
    >
      <HackerIconFill>
        <svg
          viewBox="0 0 140 140"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M60.94 82.314L17 0h20.08l25.85 52.093c.397.927.86 1.888 1.39 2.883.53.994.995 2.02 1.393 3.08.265.4.463.764.596 1.095.13.334.262.63.395.898.662 1.325 1.26 2.618 1.79 3.877.53 1.26.993 2.42 1.39 3.48 1.06-2.254 2.22-4.673 3.48-7.258 1.26-2.585 2.552-5.27 3.877-8.052L103.49 0h18.69L77.84 83.308v53.087h-16.9v-54.08z"
            fillRule="evenodd"
          />
        </svg>
      </HackerIconFill>
    </a>
  </styled.a>
</Button>
`;

exports[`HackerNews 14 1`] = `
<Button
  simpleReverse={true}
>
  <styled.a
    aria-label="Share on HackerNews"
    href="https://news.ycombinator.com/submitlink?u=&amp;t=undefined"
    rel="noreferrer noopener"
    simpleReverse={true}
    target="_blank"
  >
    <a
      aria-label="Share on HackerNews"
      className="sc-bwzfXH dqksVq"
      href="https://news.ycombinator.com/submitlink?u=&amp;t=undefined"
      rel="noreferrer noopener"
      target="_blank"
    >
      <HackerIconFill>
        <svg
          viewBox="0 0 140 140"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M60.94 82.314L17 0h20.08l25.85 52.093c.397.927.86 1.888 1.39 2.883.53.994.995 2.02 1.393 3.08.265.4.463.764.596 1.095.13.334.262.63.395.898.662 1.325 1.26 2.618 1.79 3.877.53 1.26.993 2.42 1.39 3.48 1.06-2.254 2.22-4.673 3.48-7.258 1.26-2.585 2.552-5.27 3.877-8.052L103.49 0h18.69L77.84 83.308v53.087h-16.9v-54.08z"
            fillRule="evenodd"
          />
        </svg>
      </HackerIconFill>
    </a>
  </styled.a>
</Button>
`;
